### DISCUSSION TITLE
Control task properties of `Scala run` VSCode task

### DISCUSSION THREAD

**Discussion by soronpo** [scalameta/metals-vscode#1535] [General]
When running a Scala program through the metals `run` icon, metals uses a VSCode task for that purpose.
The default configuration for the tasks prints out redundant clutter:
* In the beginning: `Executing task: .... the entire command-line ...`
* In the end: `Terminal will be reused by tasks, press any key to close it.`

Additionally, I want to clear the console at every run.
VSCode already has such options that can be configured for a task:
```
    "presentation": {
        "echo": false,    
        "showReuseMessage": false,
        "clear": true
    }
```
However, I'm not able to define such options so they take effect. 
Is there something I'm missing?

**Comment by tgodzik** [comment]
I can't seem to make that work either by adding it to task properties inside the extension. How are you trying to add it?

I wonder if this would work if we implemented https://code.visualstudio.com/api/extension-guides/task-provider

**Comment by soronpo** [comment]
I tried directly in tasks.json

**Reply by tgodzik** [reply]
That might require the task provider being defined.

**Reply by soronpo** [reply]
Could be. However vscode itself didn't highlight it as an error like it did with other methods I tried, so I thought there is a way to set these as defaults. Didn't try to make a manual task to see if indeed these take hold there.
